src/infrastructure/NamespaceHttp.ts

```diff
@@ -168,24 +167,18 @@ export class NamespaceHttp extends Http implements NamespaceRepository {
      * @returns Observable<MosaicId | null>
      */
     public getLinkedMosaicId(namespaceId: NamespaceId): Observable<MosaicId> {
-        return this.networkTypeObservable.pipe(
-            mergeMap(() =>
-                this.call(this.namespaceRoutesApi.getNamespace(namespaceId.toHex()), (body) => {
-                    const namespaceInfoDTO = body;
-                    if (namespaceInfoDTO.namespace === undefined) {
-                        // forward catapult-rest error
-                        throw namespaceInfoDTO;
+        return this.getNamespace(namespaceId).pipe(
+            mergeMap((info) => {
+                if (info.depth === 1) {
```

Comment:
Doesn't look quite right to me here. Why check level1? and also hardcoded to 1?

Reply:
As far as I can see, if the depth is > 1, the info is an alias of an alias (AB2F0159E229A0BD), and the alias type is 0

http://api-01.ap-southeast-1.0.10.0.x.symboldev.network:3000/namespaces/AB2F0159E229A0BD

If you go to the level0 (root alias E66AE40356B62BFC) you can retrieve the information of the final alias and get the real mosaic id or address

http://api-01.ap-southeast-1.0.10.0.x.symboldev.network:3000/namespaces/E66AE40356B62BFC